synapse/models/files.py

```diff
@@ -556,6 +577,54 @@ def getModelDefs(self):
                     ('base:ext', ('str', {}), {'ro': True,
                         'doc': 'The file extension.'}),
                 )),
+
+                ('file:mime:macho:loadcmd', {}, (
+                    ('file', ('file:bytes', {}), {
+                        'doc': 'The Mach-O file containing the load command.'}),
+                    ('type', ('int', {'enums': s_l_macho.getLoadCmdTypes()}), {
+                        'doc': 'The type of the load command'}),
+                    ('size', ('int', {}), {
+                        'doc': 'The size of the load command structure in bytes.'}),
+                )),
+                ('file:mime:macho:version', {}, (
+                    ('loadcmd', ('file:mime:macho:loadcmd', {}), {
+                        'doc': 'The associated load command node.'}),
+                    ('version', ('str', {}), {
+                        'doc': 'The version of the Mach-O file encoded in an LC_VERSION load command'})
+                )),
+                ('file:mime:macho:uuid', {}, (
+                    ('loadcmd', ('file:mime:macho:loadcmd', {}), {
+                        'doc': 'The actual load command '}),
+                    ('uuid', ('str', {}), {
+                        'doc': 'The UUID of the Mach-O application (as defined in an LC_UUID load command)'}),
+                )),
+                ('file:mime:macho:segment', {}, (
+                    ('loadcmd', ('file:mime:macho:loadcmd', {}), {
+                        'doc': 'The actual load command node.'}),
+                    ('loadcmd:file', ('file:bytes', {}), {
+                        'doc': 'The Mach-O file the segment is located in'}),
+                    ('name', ('str', {}), {
+                        'doc': 'The name of the Mach-O segment.'}),
+                    ('memsize', ('int', {}), {
```

Comment:
Is there an offset to the beginning of the segment we should capture here too?

Reply:
For segments yea, segment structures do come with a file offset we can record. For most segments (other than ones marked as SG_NORELOC), the file offset is a little more robust. It's not getting easily swapped out without changing many other parts of the binary.